src/directive.js

```diff
@@ -43,19 +48,20 @@
                         });
                     };
 
-                    vcRecaptcha.create(elm[0], {
-
-                        callback: callback,
-                        key: key,
-                        stoken: scope.stoken || attrs.stoken || null,
-                        theme: scope.theme || attrs.theme || null,
-                        type: scope.type || attrs.type || null,
-                        lang: scope.lang || attrs.lang || null,
-                        tabindex: scope.tabindex || attrs.tabindex || null,
-                        size: scope.size || attrs.size || null,
-                        badge: scope.badge || attrs.badge || null,
-                        'expired-callback': expired
-
+                    // Accept a promise, or resolve immediately if the value is a bare string
+                    $q.resolve(key).then(function (resolved) {
```

Comment:
IMO, this should go in the service and not in the directive. The support for key-as-promise would be complete in this project, since ATM the service doesn't support a key as promise.. Also, it could make testing the directive simpler.

Reply:
Also, should we use `$q.when` instead of `$q.resolve`? The latter was added in Angular 1.4

Reply:
You may want to have the feature in the service as well as the directive.
`$q.when` should be used for the reason mentioned above, more support.